manager/integration/tests/test_migration.py

```diff
@@ -292,22 +298,84 @@ def test_migration_with_rebuilding_replica(clients, volume_name):  # NOQA
 
 @pytest.mark.coretest  # NOQA
 @pytest.mark.migration # NOQA
-@pytest.mark.skip(reason="TODO") # NOQA
-def test_migration_with_restore_volume(clients, volume_name):  # NOQA
+def test_migration_with_restore_volume(core_api, # NOQA
+                                       client, # NOQA
+                                       clients, # NOQA
+                                       volume_name, # NOQA
+                                       storage_class, # NOQA
+                                       pvc_name, # NOQA
+                                       set_random_backupstore):  # NOQA
     """
     Test that a restored volume can be migrated.
 
     1. Prepare one backup.
     2. Create a StorageClass with `migratable` being enabled and
        `fromBackup` pointing to the above backup.
     3. Create a new RWX migratable volume using the StorageClass.
-    2. Attach to node 1, then write some data.
-    4. Attach the volume to node 2 (migration target).
-    5. Wait for the migration ready. Verify that field
+    4. Attach to node 1, then write some data.
+    5. Attach the volume to node 2 (migration target).
+    6. Wait for the migration ready. Verify that field
        `volume.controllers[0].requestedBackupRestore` is empty.
-    6. Confirm the migration then validate the data.
+    7. Confirm the migration then validate the data.
     """
-    return
+    # Step 1
+    lht_host_id = get_self_host_id()
+    volume = create_and_check_volume(client,
+                                     volume_name,
+                                     REPLICA_COUNT,
+                                     size=str(DEFAULT_VOLUME_SIZE * Gi))
+
+    volume.attach(hostId=lht_host_id)
+    volume = common.wait_for_volume_healthy(client, volume_name)
+
+    bv, b, _, _ = create_backup(client, volume_name)
```

Comment:
Do we want to write some data in the volume before taking the backup?

Reply:
Hi @khushboo-rancher , I added code to write data before take backup, thank you